test/verify/check-machines-dbus

```diff
@@ -298,6 +302,7 @@ class TestMachinesDBus(machineslib.TestMachines):
                 raise AssertionError("Unhandled distro for OVMF path")
 
             m.execute("mkdir /tmp/ovmf/; mv {0}/* /tmp/ovmf/".format(ovmf_path))
+            self.addCleanup(m.execute, "rm -rf /tmp/ovmf")
```

Comment:
This is still wrong -- the inverse of the `mv` above is to move the files back from /tmp/ovmf/ to ovmf_path. Please remove line 312 and instead put that mv into this cleanup handler.

Like that, if anything fails between here and line 312, you'll destroy the OVMF binaries.

Also, there is a much more robust way to do that: Mount a tmpfs onto ovmf_path here (or a bind mount do an empty directory), and unmount it in a cleanup handler. This will not touch /usr on disk at all, but has the same effect. I. e. something like
```
m.execute("mount -t tmpfs tmpfs " + ovmf_path)
self.addCleanup(m.execute, "umount " + ovmf_path")
```

Reply:
The line 312 is still needed, as it moves it back in and the rest of the test that can use it. Doing this is cleanup handler is too late.

But you are right, that it should also count that the `mv` part may never be executed.

I think I can just use `mount` and then do `umount` in both cleanup handler and in the test itself. (Making sure that cleanup handler does not fail if already not mounted).

Reply:
Right, I was just about to write that, I just had a look at the test again -- it needs the ovmf binaries after line 312 indeed. So addCleanup with `|| true` seems fine.